### ISSUE TITLE
Problems using pretrain weigths on a Custom Datase

### ISSUE THREAD

**Issue by DaniSerrano1** [MultimediaTechLab/YOLO#182]
I train a model with no problem using a custom dataset and custom.yml. It gave me a model_checkpoint.ckpt.

the problem is that when I import the weights in the inferences task it doesn't load. Give me this error.

I don't the problem. The path to weights is correct.

**Command**
CUDA_VISIBLE_DEVICES="0,1" python yolo/lazy.py task=inference name=inference_1 device=cuda:0 model=v9-s weight= YOLO/runs/train/v9-dev/YOLO/kvgc5lvo/checkpoints/epoch_49_step_8250.ckpt task.nms.min_confidence=0.1 task.data.source=/app/dataset/test/images task.save_predict=True hydra.verbose=True dataset=yolo_cars

**ERROR**

```
INFO     📄 Created log folder: runs/inference/inference_1                                                                                                                                                   logging_utils.py:324
[03/07/25 12:02:17] INFO     ⚡ Using 16bit Automatic Mixed Precision (AMP)                                                                                                                                                           rank_zero.py:63
                    INFO     ⚡ Trainer already configured with model summary callbacks: [<class 'yolo.utils.logging_utils.YOLORichModelSummary'>]. Skipping setting a default `ModelSummary` callback.                               rank_zero.py:63
                    INFO     ⚡ GPU available: True (cuda), used: True                                                                                                                                                                rank_zero.py:63
                    INFO     ⚡ TPU available: False, using: 0 TPU cores                                                                                                                                                              rank_zero.py:63
                    INFO     ⚡ HPU available: False, using: 0 HPUs                                                                                                                                                                   rank_zero.py:63
                    INFO     🚜 Building YOLO                                                                                                                                                                                              yolo.py:35
                    INFO       🏗  Building backbone                                                                                                                                                                                        yolo.py:38
                    INFO       🏗  Building neck                                                                                                                                                                                            yolo.py:38
                    INFO       🏗  Building head                                                                                                                                                                                            yolo.py:38
                    INFO       🏗  Building detection                                                                                                                                                                                       yolo.py:38
                    INFO       🏗  Building auxiliary                                                                                                                                                                                       yolo.py:38
                    WARNING  ⚠ Weight Not Found for key: 26.conv2.0.bottleneck.2.conv2                                                                                                                                                    yolo.py:150
                    WARNING  ⚠ Weight Not Found for key: 12.conv3.0.bottleneck.0.conv1.conv1                                                                                                                                              yolo.py:150
                    WARNING  ⚠ Weight Not Found for key: 15.conv2.0.bottleneck.1.conv1.conv2                                                                                                                                              yolo.py:150
                    WARNING  ⚠ Weight Not Found for key: 12.conv2.0.conv1                                                                                                                                                                 yolo.py:150
                    WARNING  ⚠ Weight Not Found for key: 3.conv                                                                                                                                                                           yolo.py:150

...


weight: YOLO/runs/train/v9-dev/YOLO/kvgc5lvo/checkpoints/epoch_49_step_8250.ckpt

:warning: Weight Not Found for key: 4.conv3.0.bottleneck.1.conv1.conv2
:warning: Weight Not Found for key: 12.conv3.1
:warning: Weight Not Found for key: 1

```

**Comment by LeMosquitar** [comment]
Hello I am trying to finetune this model but I am having issues,

how did you train your model? What is your custom.yaml contents and where id your dataset comefrom and its format?

**Comment by GilSeamas** [comment]
Hey @DaniSerrano1 

I have the same problem.

python yolo/lazy.py task=inference name=AnyNameYouWant device=cuda model=v9-s task.nms.min_confidence=0.1 task.fast_inference=onnx task.data.source=/home/leasemetrics/dev/YOLOX/datasets weight=path2ckpt-hfskahflsahfla.ckpt +quite=True

the code doesnt throw any error with correct path, but with incorrect path it also does not. but with correct and incorrect path to weights outputs images with no boxes.

where you able to solve it on your side?

**Comment by GilSeamas** [comment]
> Hello I am trying to finetune this model but I am having issues,
> 
> how did you train your model? What is your custom.yaml contents and where id your dataset comefrom and its format?


I think your question deviates from what DaniSerrano issue is about buuuuuuut I had some problems with coco format. so ended up following this structure:

data
 labels
  train
  validation
 images
  train
  validation

labels stores with image_name+.txt in x1,y1,x2,y2 format

custom_dataset.yaml:

path: absolute_path_to_dataset
train: train
val: validation

class_num: 2
nc: 2
class_list: ['class1', 'class2']

**Comment by GilSeamas** [comment]
> Hey [@DaniSerrano1](https://github.com/DaniSerrano1)
> 
> I have a similar problem.
> 
> python yolo/lazy.py task=inference name=AnyNameYouWant device=cuda model=v9-s task.nms.min_confidence=0.1 task.fast_inference=onnx task.data.source=/home/usr/path2data weight=path2ckpt-hfskahflsahfla.ckpt +quite=True
> 
> The code doesnt throw any error with correct path, but with incorrect path it also does not. Both with correct and incorrect path to weights the outputs images have no boxes.
> 
> were you able to solve it on your side?

this solved my issue @DaniSerrano1

**Comment by divyanidhi** [comment]
Hi @GilSeamas 
I am trying to use this #136 comment PR, but the inference on my custom.png is trying to map 80 classes (referring to coco, I think).  the custom fine tuned model that I am trying to use has only 7 num of class. Any suggestions for this?

Can you also please share your run command for inference?
